c2pa_c_ffi/src/c_api.rs

```diff
@@ -955,18 +974,96 @@ pub unsafe extern "C" fn c2pa_builder_add_action(
 ) -> c_int {
     check_or_return_int!(builder_ptr);
     let mut builder = guard_boxed_int!(builder_ptr);
-    let action_json = from_cstr_or_return_int!(action_json);
 
     // Parse the JSON into a serde Value to use with the Builder
-    let action_value: serde_json::Value = match serde_json::from_str(&action_json) {
+    let action_value = match parse_json_from_cstr(action_json) {
         Ok(value) => value,
+        Err(err) => return err,
+    };
+
+    match builder.add_action(action_value) {
+        Ok(_) => 0, // returns 0 on success
         Err(err) => {
-            Error::from_c2pa_error(c2pa::Error::JsonError(err)).set_last();
-            return -1;
+            Error::from_c2pa_error(err).set_last();
+            -1
         }
+    }
+}
+
+/// Adds an assertion to the manifest.
+/// At the C-level of the interface, we will only get JSON as data,
+/// and layers above will handle the conversion to the appropriate
+/// JSON structure.
+///
+/// # Parameters
+/// * builder_ptr: pointer to a Builder.
+/// * label: A label for the assertion (null-terminated C string).
+/// * data_json: The data for the assertion as a JSON string (null-terminated C string).
+///
+/// # Errors
+/// Returns -1 if there were errors, otherwise returns 0.
+/// The error string can be retrieved by calling c2pa_error.
+///
+/// # Safety
+/// Reads from NULL-terminated C strings.
+#[no_mangle]
+pub unsafe extern "C" fn c2pa_builder_add_assertion(
```

Comment:
With great power comes great responsibility.
Should we even expose that here? I think it makes sense to expose it since it's a layer deep down, and defer the responsibility of the call to the caller.